### PR TITLE
fix(swap): Fix swap cancellation fee error handling

### PR DESCRIPTION
**Pull request by michalszorad**
resolves https://emurgo.atlassian.net/browse/YOMO-881

This PR only improves error handling, however there may be cases when the REST API does not respond with CBOR such as in this example 
https://aggregator.muesliswap.com/cancelSwapTransaction/?wallet=01417232dba248bdd316442f323c4279fb1e0ef7bfb16680d8dabdd655f9ea671653e35395356465e4bd8f563b5d5480e4eb466da896a22afc&utxo=8956d68753d718afbaafde0e83dc1cb1d205da3c89fb08c924ab1d63fd953ed2%230&collateralUtxo=82825820926f781eea85507363790aca75e4e4a05ff6a0ad1c8b823db4074fd515b521a80082583901417232dba248bdd316442f323c4279fb1e0ef7bfb16680d8dabdd655f9ea671653e35395356465e4bd8f563b5d5480e4eb466da896a22afc1a001e8480

### CODE REVIEW

**Review by jorbuedo — APPROVED** [review]

**Comment by jorbuedo** [comment]
Lgtm. API said that order in particular was faulty and that's why cancellation doesn't work.

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx [hunk 1]
```diff
@@ -194,19 +194,12 @@ export const OpenOrders = () => {
...
-      let fee = '0'
       setIsLoading(true)
-
-      try {
-        fee = await getCancellationOrderFee(wallet, cancelOrder, {orderUtxo: utxo, collateralUtxo, bech32Address})
-      } catch (error) {
-        Alert.alert(strings.generalErrorTitle, strings.generalErrorMessage(error))
-      }
-
+      const fee = await getCancellationOrderFee(wallet, cancelOrder, {orderUtxo: utxo, collateralUtxo, bech32Address})
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx:198]: Without the try/catch Alert, where is it dealing with the missing cbor throw?

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx:198]: In the `openCancellationModal`

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx [hunk 2]
```diff
@@ -230,24 +223,29 @@ export const OpenOrders = () => {
...
+          assetToLabel={assetToLabel}
+          assetAmount={`${tokenAmount} ${assetToLabel}`}
+          assetPrice={`${tokenPrice} ${assetFromLabel}`}
+          totalReturned={totalReturned}
+          fee={fee}
+        />,
+        460,
+      )
+    } catch (error) {
+      if (error instanceof Error) {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx:245]: is this if required?

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx:245]: Yes, error is `unknown` in the catch and we would not be able to access `error.message`

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx:245]: in that case we need also an alert on the else side, if somewhere deep something else is thrown.

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx [hunk 3]
```diff
@@ -230,24 +223,29 @@ export const OpenOrders = () => {
...
+        Alert.alert(strings.generalErrorTitle, strings.generalErrorMessage(error.message))
+      }
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx]:
```suggestion
        Alert.alert(strings.generalErrorTitle, strings.generalErrorMessage(error.message))
      } else {
        Alert.alert(strings.generalErrorTitle, strings.generalErrorMessage(/* stringify the error - default error unknown? */))
      }
```

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx]: smth alike

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/OpenOrders.tsx]: That's now updated

**Pull request merged**